### ISSUE TITLE
Adding a local named region results in a corrupt excel file

### ISSUE THREAD

**Issue by mrainers** [JanMarvin/openxlsx2#829]
After I added a named region with the parameter `local_sheet = TRUE` to my workbook, I get an error message when opening it in excel and the named region is removed from the file.

Example to reproduce:
```r
wb <- wb_workbook()
wb$add_worksheet()

wb$add_data(x = "bar")
wb$add_named_region(
  name = "foo",
  dims = "A1",
  local_sheet = TRUE
)

wb$open()
```

`wb$get_named_regions()` prints:
```r
  localSheetId name               value  sheets coords id local sheet
1            1  foo 'Sheet 1'!$A$1:$A$1 Sheet 1  A1:A1  1     1     1
```

### Version
openxlsx2   * 1.0.0.9000 2023-10-11 [1] Github (JanMarvin/openxlsx2@419abdc)
UPDATE: also affects current version: 
openxlsx2   * 1.1     2023-10-23 [1] Github (JanMarvin/openxlsx2@ee9edb4)

### Session info 
```
 setting  value
 version  R version 4.3.1 (2023-06-16 ucrt)
 os       Windows 10 x64 (build 19045)
 system   x86_64, mingw32
 ui       RStudio
 language (EN)
 collate  German_Germany.utf8
 ctype    German_Germany.utf8
 tz       Europe/Berlin
 date     2023-10-23
 rstudio  2023.09.0+463 Desert Sunflower (desktop)
 pandoc   NA
```

**Comment by JanMarvin** [comment]
Hi @mrainers , thanks for the report! Our counting is wrong. The sheet index is zero based.

```R
library(openxlsx2)

wb <- wb_workbook()
wb$add_worksheet()

wb$add_data(x = "bar")
wb$add_named_region(
  name = "foo",
  dims = "A1",
  local_sheet = TRUE
)


wb$workbook$definedNames
wb$workbook$definedNames <- "<definedName localSheetId=\"0\" name=\"foo\">'Sheet 1'!$A$1:$A$1</definedName>"

wb$open()
```

**Comment by JanMarvin** [comment]
A fix is in #830. Unfortunately, this morning I've submitted release 1.1 to CRAN, therefore the fix will have to wait a little

**Comment by mrainers** [comment]
Thx for the really quick fix :-)
